### DISCUSSION TITLE
best pimd configuration on linux

### DISCUSSION THREAD

**Discussion by lucdig** [troglobit/pimd#239] [Q&A]
Hi, I have a couple of question about pimd

1) if I've understood correctly, the RP router of an AS should be connected with a link (physical or tunnel) to all the other no-RP multicast routers running pimd, is it correct? Or, can the RP be reached through a route between other multicast routers of the same AS?

2) I've seen the blog post https://troglobit.com/howtos/pimd-on-openbsd/, and I would like to know, in your opinion, what is the best way of running pimd/ripd on Linux. I see that there are several options (quagga/zebra, frr, ...) and I'm quite lost

Thanks, regards.
Luca

**Comment by troglobit** [comment]
1. The RP is used only initially, as a distribution point for all multicast.  When receivers are registered a shortest-path tree (SPT) is formed and closer routers are used for each receiver.  If not all routers in the AS run PIM they need to be connected using a tunnel.
2. It's all very simple.  After `gated` was abandoned, Zebra arrived on the scene as the primary Open Source routing suite for Linux (and *BSD).  It became GNU Zebra and then the maintainer had a falling-out with FSF and since he owned the copyright, he founded IP Infusion and used Zebra as the baseline for ZebOS.  All development on GNU Zebra eventually stopped, but it was forked to a new project Quagga.  Quagga development continued but eventually slowed and after another schism was forked again into Frr.  As an homage to the original Zebra, the daemon responsible for muxing all routes to/from the kernel and the different routing daemons remains named `zebra`.  Hence the confusion.  To get RIP going you can use either Zebra, Quagga, or Frr.  But also an alternative routing suite called BIRD.  The latter is a very tight and nice re-implementation of most of the common unicast routing protocols.  So if you want to use pimd, BIRD RIP could provide a small and simple setup.   However, if you just want to run PIM and don't care that much about size, I'd recommend starting with Frr.  It has its own PIM implementation that is well maintained and well integrated with the other routing daemons.

**Reply by lucdig** [reply]
Thanks, BIRD looks nice.
I've seen that pimd doesn't run on OpenBSD (7.2 GENERIC). After compiling, the daemon doesn't start because:

`Cannot set PIM flag in kernel: Protocol not available`

I can't test pimd/ripd because I work with OpenBSD as well as with Linux.

Regards.

**Reply by troglobit** [reply]
Unfortunately, OpenBSD have removed PIM support from the MROUTING stack in their kernel. I do not know from which version but it's been a while now.
